packages/app/src/app/pages/Sandbox/Editor/Workspace/screens/ProjectInfo/Title.tsx

```diff
@@ -0,0 +1,84 @@
+import { getSandboxName } from '@codesandbox/common/lib/utils/get-sandbox-name';
+import { ESC, ENTER } from '@codesandbox/common/lib/utils/keycodes';
+import React, { ChangeEvent, KeyboardEvent, useState } from 'react';
+
+import { useOvermind } from 'app/overmind';
+
+import { Text, Input, Stack } from '@codesandbox/components/lib';
+import styled from 'styled-components';
+import { PenIcon } from './icons';
+
+const Icon = styled(PenIcon)`
+  cursor: pointer;
+  display: none;
+`;
+
+const Name = styled(Stack)`
+  :hover {
+    svg {
+      display: block;
+    }
+  }
+`;
+
+export const Title = ({ editable }) => {
+  const {
+    actions: {
+      workspace: { sandboxInfoUpdated, valueChanged },
+    },
+    state: {
+      editor: { currentSandbox },
+      workspace: {
+        project: { title },
+      },
+    },
+  } = useOvermind();
+  const [editing, setEditing] = useState(false);
+
+  const onChange = ({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+    valueChanged({ field: 'title', value });
+  };
+
+  const onKeyUp = ({ keyCode }: KeyboardEvent<HTMLInputElement>) => {
+    if (keyCode === ENTER) {
+      sandboxInfoUpdated();
+
+      setEditing(false);
+    } else if (keyCode === ESC) {
+      setEditing(false);
+    }
+  };
+
+  const onBlur = () => {
+    sandboxInfoUpdated();
+    setEditing(false);
+  };
+
+  const edit = () => {
+    valueChanged({
+      field: 'title',
+      value: getSandboxName(currentSandbox),
+    });
+
+    setEditing(true);
+  };
+
+  return editing ? (
+    <Stack>
```

Comment:
You don't need Stack here. The input should be full width (You probably want to wrap it a SidebarRow and FormField though

Reply:
Wrapped in FormField,